protocol/rep/rep.go

```diff
@@ -227,9 +225,8 @@ func (r *rep) RecvHook(m *mangos.Message) bool {
 	}
 	r.sock.SetSendError(nil)
 	r.backtraceL.Lock()
-	r.backtrace = append(r.backtracebuf[0:0], m.Header...)
+	r.backtrace = append(r.backtrace[0:0], m.Header...)
 	r.backtraceL.Unlock()
-	m.Header = nil
```

Comment:
I'm not at all confident this is correct.  That said, I understand your desire for further introspection into what REP is doing, and lost messages in particular.  I need to think about what you're doing here carefully, as I think there may be implications for device chains.  In particular, we need to make sure that we use the saved backtrace rather than a user-supplied header.  Clearing the header is how we make this work.

It may be better instead to have a new API to access the backtrace buffer.  I'm not sure.